drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -8188,10 +8223,16 @@ void RenderingDeviceVulkan::compute_list_end(BitField<BarrierMask> p_post_barrie
 		barrier_flags |= VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT;
 		access_flags |= VK_ACCESS_SHADER_READ_BIT | VK_ACCESS_SHADER_WRITE_BIT;
 	}
-	if (p_post_barrier.has_flag(BARRIER_MASK_RASTER)) {
-		barrier_flags |= VK_PIPELINE_STAGE_VERTEX_INPUT_BIT | VK_PIPELINE_STAGE_VERTEX_SHADER_BIT | VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT | VK_PIPELINE_STAGE_DRAW_INDIRECT_BIT;
+	if (p_post_barrier.has_flag(BARRIER_MASK_VERTEX)) {
+		// Q: should we remove VK_PIPELINE_STAGE_DRAW_INDIRECT_BIT + VK_ACCESS_INDIRECT_COMMAND_READ_BIT here?
```

Comment:
No, as we may issue indirect draw commands for SDFGI and VoxelGI. 

For now, let's pair it with vertex and fragment as it seems we don't get any granularity. We don't use draw indirect often, and it won't be any worse than the current case, so I think keeping it in is the safest choice

Reply:
ok, I wasn't aware of the use case here, I should really look into that.

Agree that for now it's best to keep it as is.